src/strategy.ts

```diff
@@ -53,66 +53,83 @@ export abstract class AbstractStrategy extends PassportStrategy {
       profile,
       loggedOut,
     }: {
-      profile?: Profile | null;
-      loggedOut?: boolean;
+      profile: Profile | null;
+      loggedOut: boolean;
     }) => {
       if (loggedOut) {
+        // Log out the current user no matter if we can verify the logged in user === logout requested user
         req.logout();
-        if (profile) {
+
+        // Check to see if we are logging out the user that is currently logged in to craft a proper IdP response
+        if (profile != null) {
           if (this._saml == null) {
             throw new Error("Can't get logout response URL without a SAML provider defined.");
           }
 
-          const RelayState =
-            (req.query && req.query.RelayState) || (req.body && req.body.RelayState);
-          return this._saml.getLogoutResponseUrl(profile, RelayState, options, redirectIfSuccess);
-        }
-        return this.pass();
-      }
-
-      const verified = (
-        err: Error | null,
-        user?: Record<string, unknown>,
-        info?: Record<string, unknown>
-      ) => {
-        if (err) {
-          return this.error(err);
+          const RelayState = req.query?.RelayState || req.body?.RelayState;
+
+          if (
+            req.user != null &&
+            req.user.ID === profile.ID &&
```

Comment:
Isn't this `ID` (e.g. `profile.ID`) just a random string associated to SAML protocol messages (and `profile.ID` would be random string associated to `LogoutRequest`)

Reply:
You are correct. Adjusted.